### PR TITLE
Allow arbitrary expression evaluation in chromeinspector

### PR DESCRIPTION
**Pull request by Akirathan**
### Pull Request Description

Allow arbitrary expression evaluation in the chromeinspector console. Moreover, allow modifications of any variable in any stack frame.

### Important Notes

- Implement inline parsing in `EnsoLanguage.parse(InlineParsingRequest)`.
- Debugging experience is affected by this [bug in Truffle](https://github.com/oracle/graal/issues/5513), which causes NPEs when a host object gets into chromeinspector. I tried to implement a workaround, but it does not work all the time. Nevertheless, it should not matter that much - if there is a NPE in the debugger, you can just ignore it, as it should be concealed in the debugger and should not be propagted outside. See comments in the `docs/debugger`.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/interpreter/EnsoLanguage.java [hunk 1]
```diff
@@ -174,6 +189,117 @@ protected CallTarget parse(ParsingRequest request) {
...
+      Tree inlineExpr = context.getCompiler().parseInline(request.getSource());
+      var undesirableExprTypes = List.of(
+          Tree.Assignment.class,
+          Tree.Import.class,
+          Tree.Export.class
+      );
+      if (astContainsExprTypes(inlineExpr, undesirableExprTypes)) {
+        throw new InlineParsingException(
+            "Inline parsing request contains some of undesirable expression types: " + undesirableExprTypes,
+            null
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/interpreter/EnsoLanguage.java]: include `inlineExpr` in the exception

**Review by hubertp — APPROVED** [review]
lgtm

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java [hunk 2]
```diff
@@ -89,6 +90,76 @@ IR.Module translate(Tree ast) {
...
+   * to the compiler's {@link IR} representation.
+   *
+   * Inline expressions must <b>only</b> be expressions, and may not contain any
+   * type of definition.
+   *
+   * @param ast The tree representing the expression to translate.
+   * @return The {@link IR} representation of the given ast if it is valid, otherwise
+   *  {@link Option#empty()}.
+   */
+  Option<IR.Expression> translateInline(Tree ast) {
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:104]: Great, thank you.

File: engine/runtime/src/main/java/org/enso/interpreter/instrument/HostObjectDebugWrapper.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import java.util.Arrays;
+import org.enso.interpreter.EnsoLanguage;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+
+/**
+ * Wrapper for host objects. Is a workaround for a bug in chromeinspector
+ * (https://github.com/oracle/graal/issues/5513). All the host objects should
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/instrument/HostObjectDebugWrapper.java:18]: We should push towards fixing the issue in GraalVM. I'd like the fix to be in 23.0 version.

**Review by JaroslavTulach — APPROVED** [review]

**Pull request merged**